src/common/src/types/mod.rs

```diff
@@ -207,6 +230,20 @@ impl From<&ProstDataType> for DataType {
 }
 
 impl DataType {
+    pub const BOOLEAN: DataType = DataType::Boolean;
```

Comment:
Why we need this?

Reply:
https://docs.rs/postgres-types/latest/src/postgres_types/type_gen.rs.html#891

Reply:
I am still confused on why we have this. It is leading to inconsistent styles: `DataType::Date` and `DataType::DATE` can be used interchangeably. (e.g. #6738)

Reply:
DataType::Date is a type. DataType::DATE is a global instance. But in indeed it can be used interchangeably in 'match'😢. Maybe we should remove it. @BowenXiao1999

Reply:
> DataType::Date is a type

It's not a type. It's a match pattern instead. But yes, we can use both for `match`.